src/Framework.Cqrs/Mediator/IMediator.cs

```diff
@@ -8,9 +9,9 @@
 
     public interface IMediator
     {
-        TResponse Query<TResponse>(IQuery<TResponse> query);
+        IEnumerable<Response<TResponse>> Query<TQuery, TResponse>(TQuery query) where TQuery : IQuery<TResponse>;
 
-        Task<TResponse> QueryAsync<TResponse>(IQuery<TResponse> query);
+        Task<IEnumerable<Response<TResponse>>> QueryAsync<TQuery, TResponse>(TQuery query) where TQuery : IQuery<TResponse>;
```

Comment:
You missed one huge disadvantage: performance :)

Reply:
I know it can be 400x slower. But it also has its uses. In this case was to solve a problem easily 😄 

I don't know how worse the performance would be if we used it. I just don't think performance may be the only argument to avoid using it.

This is a good discussion 😄

Reply:
Let's discuss this later when we are done with our MVP. I'm not against reflection if we keep it to a minimum and we implement some kind of cache mechanism so that we don't have to use reflection every time we want to execute a query or command.